lighthouse-core/computed/trace-of-tab.js

```diff
@@ -43,7 +43,8 @@ class TraceOfTab {
    * @return {Promise<LH.Artifacts.TraceOfTab>}
   */
   static async compute_(trace) {
-    return LHTraceProcessor.computeTraceOfTab(trace);
+    const traceOfTab = await LHTraceProcessor.computeTraceOfTab(trace, {throwOnNoFCP: true});
+    return /** @type {LH.Artifacts.TraceOfTab} */ (traceOfTab);
```

Comment:
I really don't like this, but I was blanking on a way to do this without casting and/or manually reconstructing the entire object. 

any ideas @brendankenny ?

Reply:
what if we didn't throw in `trace-processor.js` at all and only throw `NO_FCP` out here if it's not on the return value?

Reply:
doesn't that involve manually reconstructing every object that has fcp?

Reply:
> doesn't that involve manually reconstructing every object that has fcp?

yeah, I can't think of a way around it, tsc won't track the validations on all three existence checks into the return value :/

It would allow simplifying `trace-processor.js` (no argument, no fcp check in there, etc), though, and it would keep the fcp checks and cast in the same place in this file.